apps/mobile/src/screens/PostScreen/SharePostBottomSheet.tsx

```diff
@@ -0,0 +1,294 @@
+import { useBottomSheetDynamicSnapPoints } from '@gorhom/bottom-sheet';
+import Clipboard from '@react-native-clipboard/clipboard';
+import { useColorScheme } from 'nativewind';
+import { ReactElement, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { Linking, View } from 'react-native';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+import CopyIcon from 'src/icons/CopyIcon';
+import FarcasterIcon from 'src/icons/FarcasterIcon';
+import LensIcon from 'src/icons/LensIcon';
+import { TwitterIcon } from 'src/icons/TwitterIcon';
+import { noop } from 'swr/_internal';
+
+import { Button } from '~/components/Button';
+import { FadedInput } from '~/components/FadedInput';
+import {
+  GalleryBottomSheetModal,
+  GalleryBottomSheetModalType,
+} from '~/components/GalleryBottomSheet/GalleryBottomSheetModal';
+import { useSafeAreaPadding } from '~/components/SafeAreaViewWithPadding';
+import { Typography } from '~/components/Typography';
+import { SharePostBottomSheetQuery } from '~/generated/SharePostBottomSheetQuery.graphql';
+import { contexts } from '~/shared/analytics/constants';
+import { getPreviewImageUrlsInlineDangerously } from '~/shared/relay/getPreviewImageUrlsInlineDangerously';
+import colors from '~/shared/theme/colors';
+
+import MiniPostOpenGraphPreview from './MiniPostOpenGraphPreview';
+
+const SNAP_POINTS = ['CONTENT_HEIGHT'];
+
+type Props = {
+  postId: string;
+  title?: string;
+  creatorName?: string;
+  onClose?: () => void;
+};
+
+export function SharePostBottomSheet({ title, creatorName, postId, onClose }: Props) {
+  const queryResponse = useLazyLoadQuery<SharePostBottomSheetQuery>(
+    graphql`
+      query SharePostBottomSheetQuery($postId: DBID!) {
+        post: postById(id: $postId) {
+          ... on ErrPostNotFound {
+            __typename
+          }
+          ... on Post {
+            __typename
+            author @required(action: THROW) {
+              username
+              profileImage {
+                ... on TokenProfileImage {
+                  token {
+                    ...getPreviewImageUrlsInlineDangerouslyFragment
+                  }
+                }
+                ... on EnsProfileImage {
+                  __typename
+                  profileImage {
+                    __typename
+                    previewURLs {
+                      medium
+                    }
+                  }
+                }
+              }
+            }
+            caption
+            tokens {
+              name
+              ...getPreviewImageUrlsInlineDangerouslyFragment
+            }
+          }
+        }
+      }
+    `,
+    { postId: postId }
+  );
+
+  const { post } = queryResponse;
+  const { bottom } = useSafeAreaPadding();
+  const [hasCopiedUrl, setHasCopiedUrl] = useState(false);
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType>(null);
+  useEffect(() => bottomSheetRef.current?.present(), []);
```

Comment:
Can we used `forwardRef` for this bottom sheet component? So, we don't the useEffect and we can control (open/close) it from the parent component. 

If we wanna go with the forwardRef, we might need to move the Suspense inside this bottom sheet. 

There is an example how we used it at `WarningPostBottomSheet.tsx`